spectree/plugins/flask_plugin.py

```diff
@@ -122,30 +122,34 @@ def parse_path(self, route, path_parameter_descriptions):
 
         return "".join(subs), parameters
 
-    def request_validation(self, request, query, json, form_data, headers, cookies):
+    def request_validation(self, request, query, json, form, headers, cookies):
         req_query = request.args or {}
-        if request.mimetype in self.FORM_MIMETYPE:
-            req_json = request.form or {}
-
-            for key, file in request.files.items():  # TODO: My eyes bleed. How to improve this loop?
-                req_json = dict(
-                    list(request.form.items())
-                )
-                req_json.update({key: {
-                    "filename": file.filename,
-                    "name": file.name,
-                    "content_type": file.content_type,
-                    "stream": file.stream.read(),
-                    "content_length": file.stream.tell(),
-                }})
-        else:
-            req_json = request.get_json(silent=True) or {}
+        req_json = request.get_json(silent=True) or {}
+        req_form = request.form or {}  # or MultiDict() ?
         req_headers = request.headers or {}
         req_cookies = request.cookies or {}
+
+        if request.mimetype in self.FORM_MIMETYPE:
+            body = form or json or None
+            if body:
+                form_dict = {}
+                for k, v in body.schema().get('properties', {}).items():
+                    if v.get('type') == 'array':
+                        value = req_form.getlist(k)
+                    else:
+                        value = req_form.get(k)
+                    if value is not None:
+                        form_dict[k] = value
+                form_dict.update(**request.files.to_dict())
+                if form:
+                    req_form = form_dict
+                elif json:
```

Comment:
maybe if here?

Reply:
elsewhere in this PR there is a condition that will raise exception if both `json` and `form` exist, so it seems like they are mutually exclusive